File: modulesCore/homeScreen/src/main/java/cn/pivotstudio/modulec/homescreen/ui/adapter/AllForestAdapter.java
package cn.pivotstudio.modulec.homescreen.ui.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;
import cn.pivotstudio.modulec.homescreen.databinding.ItemAllForestBinding;
import cn.pivotstudio.modulec.homescreen.model.ForestResponse;

public class AllForestAdapter extends ListAdapter<ForestResponse.ForestCard, AllForestAdapter.ForestCardViewHolder> {

    private View.OnClickListener itemClickListener;

    public static final DiffUtil.ItemCallback<ForestResponse.ForestCard> DIFF_CALLBACK = new DiffUtil.ItemCallback<ForestResponse.ForestCard>() {
        @Override
        public boolean areItemsTheSame(@NonNull ForestResponse.ForestCard oldItem, @NonNull ForestResponse.ForestCard newItem) {
            return false;
        }

        @Override
        public boolean areContentsTheSame(@NonNull ForestResponse.ForestCard oldItem, @NonNull ForestResponse.ForestCard newItem) {
            return false;
        }
    };

    public AllForestAdapter(View.OnClickListener clickListener) {
        super(DIFF_CALLBACK);
        itemClickListener = clickListener;
    }

    @NonNull
    @Override
    public ForestCardViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ForestCardViewHolder(
                ItemAllForestBinding.inflate(LayoutInflater.from(parent.getContext()),
                        parent,
                        false)
        );
    }

    @Override
    public void onBindViewHolder(@NonNull ForestCardViewHolder holder, int position) {
        ForestResponse.ForestCard card = getItem(position);
        holder.itemView.setOnClickListener(itemClickListener);
        holder.bind(card);
    }

    static class ForestCardViewHolder extends RecyclerView.ViewHolder {
        private final ItemAllForestBinding binding;

        public ForestCardViewHolder(ItemAllForestBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        void bind(ForestResponse.ForestCard forestCard) {
            binding.setForestCard(forestCard);
        }

    }

}

File: modulesCore/homeScreen/src/main/java/cn/pivotstudio/modulec/homescreen/ui/adapter/AllForestAdapter.kt
package cn.pivotstudio.modulec.homescreen.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import cn.pivotstudio.husthole.moduleb.network.model.ForestBrief
import cn.pivotstudio.modulec.homescreen.databinding.ItemAllForestBinding

class AllForestAdapter(
    val onItemClick: (String) -> Unit
) : ListAdapter<Pair<String, List<ForestBrief>>, AllForestAdapter.AllForestViewHolder>(DiffCallback) {

    inner class AllForestViewHolder(
        private var binding: ItemAllForestBinding,
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(forestBriefList: Pair<String, List<ForestBrief>>) {
            val adapter = AllForestItemAdapter(onItemClick)
            binding.itemRecyclerView.adapter = adapter
            binding.type = forestBriefList.first
            adapter.submitList(forestBriefList.second)
        }

    }

    companion object DiffCallback : DiffUtil.ItemCallback<Pair<String, List<ForestBrief>>>() {
        override fun areItemsTheSame(
            oldItem: Pair<String, List<ForestBrief>>,
            newItem: Pair<String, List<ForestBrief>>
        ): Boolean {
            return oldItem.first == newItem.first
        }

        override fun areContentsTheSame(
            oldItem: Pair<String, List<ForestBrief>>,
            newItem: Pair<String, List<ForestBrief>>
        ): Boolean {
            return false
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AllForestViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        return AllForestViewHolder(
            ItemAllForestBinding.inflate(layoutInflater, parent, false),
        )
    }

    override fun onBindViewHolder(holder: AllForestViewHolder, position: Int) {
        val briefs = getItem(position)
        holder.bind(briefs)
    }

}